docs/source-2.0/guides/generating-code/generating-a-client.rst

```diff
@@ -0,0 +1,161 @@
+    │   ├── smithyprojections
+    │   │   └── weather-service
+    │   │       └── source
+    │   │           ├── build-info/
+    │   │           ├── model/
+    │   │           ├── sources/
+    │   │           └── typescript-codegen
+    │   │               ├── LICENSE
+    │   │               ├── package.json
+    │   │               ├── src/
+    │   │               ├── tsconfig.cjs.json
+    │   │               ├── tsconfig.es.json
+    │   │               ├── tsconfig.json
+    │   │               ├── tsconfig.types.json
+    │   │               └── typedoc.json
+    │   └── tmp
+    ├── build.gradle.kts
+    ├── model
+    │   └── weather.smithy
+    └── smithy-build.json
+
+
+Using the generated code
+========================
+
+The generated code is just a normal TypeScript package. Each time the model
+is built and the code generated, the TypeScript code also has to be compiled.
+The generated ``package.json`` contains scripts to do so:
+
+.. code-block:: json
+
+    "scripts": {
+        "build": "concurrently 'yarn:build:cjs' 'yarn:build:es' 'yarn:build:types'",
+        "build:cjs": "tsc -p tsconfig.cjs.json",
+        "build:docs": "typedoc",
+        "build:es": "tsc -p tsconfig.es.json",
+        "build:types": "tsc -p tsconfig.types.json",
+        "build:types:downlevel": "downlevel-dts dist-types dist-types/ts3.4",
+        "clean": "rimraf ./dist-* && rimraf *.tsbuildinfo",
+        "prepack": "yarn run clean && yarn run build"
+    }
+
+This example creates a mono-repo using `Yarn Workspaces`_ that
+integrates building the Smithy model and generating the code into the
+development workflow. First, move the Smithy project into its own
+directory named ``smithy/``::
+
+    .
+    └── smithy
+        ├── build
+        ├── build.gradle.kts
+        ├── model
+        └── smithy-build.json
+
+Next, create a ``package.json`` in the root of the project with the following
+contents:
+
+.. code-block:: json
+
+    {
+      "name": "weather-service",
+      "scripts": {
+        "generate": "cd smithy && gradle clean build",
+        "build": "yarn workspace @weather-service/client build",
+      },
+      "dependencies": {
+        "@weather-service/client": "0.0.1"
+      },
+      "private": true,
+      "workspaces": [
+        "smithy/build/smithyprojections/smithy/client/typescript-codegen"
+      ]
+    }
+
+A few things to note:
+
+* The path under ``workspaces`` is the path to the root of the generated
+  TypeScript package.
+* The ``build`` script compiles the generated TypeScript package,
+  referred to by the name specified in the ``typescript-codegen`` plugin
+  configuration in ``smithy-build.json``.
+* A ``generate`` script which builds the model, re-generating the code.
+* A dependency has been added on the generated TypeScript package, using the
+  name and version specified in the ``typescript-codegen`` plugin configuration
+  in ``smithy-build.json``
+
+Finally, create an ``app.ts`` file to use the client:
```

Comment:
The series of commands to build the workspace is not clear (e.g. is `generate` or `build` called first)?